### PR TITLE
Extending more functionality to the matrix module

### PR DESCRIPTION
**Pull request by PulkitMishra**
1. Adds Diagonal Matrix Class
2. Identity Matrix Class now inherits from the Diagonal Matrix class
3. update gitignore and minor doc fixes

### CODE REVIEW

**Comment by PulkitMishra** [comment]
@8hantanu please have a look. Apology for not updating the previous PR and creating a new one instead. Wanted the working tree clean and didn't want to put in too much labour and time in it (Exam tomorrow) so picked the easy way out and went for a hard reset (since it involved only 3 files)
All pointers raised in the previous review have been taken care of.

File: visma/matrix/structure.py [hunk 1]
```diff
@@ -103,22 +131,42 @@ def determinant(self):
...
+    """Class for identity matrix
+
+    Identity matrix is a diagonal matrix with all elements as 0 except for the diagonal elements which are 1.
+
+    Extends:
+        DiagMat
+    """
+
+    def __init__(self, dim):
+        super().__init__(dim, ["1"]*dim[0])
```
Comment (8hantanu) [inline visma/matrix/structure.py]: Remove lines 171-172. Idenmat is created from 171 itself.
```suggestion
        super().__init__(dim, [[Constant(1)]]*dim[0])
```

File: tests/test_matrix.py [hunk 2]
```diff
@@ -14,6 +15,67 @@ def test_strMatrix():
...
+    mat = getTokens("[1+x, 0+y; \
+                      0, z]")
+    assert not mat.isDiagonal()
+
+    mat = getTokens("[1, 2; \
+                      1, 3; \
+                      1, 4]")
+    assert not mat.isDiagonal()
+
+    mat = DiagMat([3, 3], [['1'], ['5'], ['2']])
```
Comment (8hantanu) [inline tests/test_matrix.py]:
```suggestion
    mat = DiagMat([3, 3], [[Constant(1)], [Constant(5)], [Constant(2)]])
```

**Review by 8hantanu — COMMENTED** [review]
You can update this after your exams :)

**Comment by PulkitMishra** [comment]
> You can update this after your exams :)

Changes made. And Nah .. This is wayyyy more enjoyable.
I mean I dont really care much about them. Getting 9+ is easy and I dont want more than that. I'd rather focus on building my skills.

**Review by 8hantanu — APPROVED** [review]

**Pull request merged**